### PR TITLE
Conformance profiles summary and motivation

### PR DESCRIPTION
**Pull request by johnbelamaric**

### CODE REVIEW

**Comment by johnbelamaric** [comment]
This is a start on the profiles KEP, still a WIP of course.

/cc @WilliamDenniss @spiffxp

**Comment by johnbelamaric** [comment]
/hold
for all approvals (also still a WIP)

**Comment by hh** [comment]
I think this gets us to our initial stages.
/lgtm

File: keps/sig-architecture/1618-conformance-profiles/README.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+
+#### Role: End User
+
+As an end user deploying an application on Kubernetes, I must be able to
+identify the profiles supported by my cluster and the application I wish to
+deploy.
+
+ - Should be able to define which tests to run up-front (eg: conformance, plus
+   profiles, profiles only, etc)
+ - May be able to detect or enumerate list of eligible profiles for a given
```
Comment (derekwaynecarr) [inline keps/sig-architecture/1618-conformance-profiles/README.md:268]: is this via some query-able endpoint under discussion?

Comment (hh) [reply keps/sig-architecture/1618-conformance-profiles/README.md:268]: The idea of polling for features would be great.

Comment (Jefftree) [reply keps/sig-architecture/1618-conformance-profiles/README.md:268]: We haven't ironed out the technical details yet, but this is a great suggestion.

File: keps/sig-architecture/1618-conformance-profiles/README.md [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+of profiles that define broad functional boundaries, applications will be able
+to document the profiles required in order for them to run.
+
+### Goals
+
+ * Design and implement a mechanism to express profiles in the conformance
+   behavior definitions.
+ * Design and implement a mechanism to validate whether a cluster conforms to
+   each profile.
+ * Document guidance on how to choose the scope of individual profiles.
```
Comment (spiffxp) [inline keps/sig-architecture/1618-conformance-profiles/README.md:209]: and restrict the total number of profiles?

File: keps/sig-architecture/1618-conformance-profiles/README.md [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Role: Developer
+
+As a developer of an optional feature, I must be able to define behaviors for
+that feature that a vendor claiming support for the feature must follow.
+
```
Comment (spiffxp) [inline keps/sig-architecture/1618-conformance-profiles/README.md:244]: Whose responsibility is it to assign optional features to Profiles / create new Profiles?  

I feel like as a developer I need to know this information if I want my feature to be included in Conformance testing.  

OTOH, maybe inclusion in a Profile is up to someone else to decide

File: keps/sig-architecture/1618-conformance-profiles/README.md [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+deploy.
+
+ - Should be able to define which tests to run up-front (eg: conformance, plus
+   profiles, profiles only, etc)
+ - May be able to detect or enumerate list of eligible profiles for a given
+   cluster (eg: it looks like you have a cloud provider configured, it looks
+   like you have a storage driver, etc)
+ - Need to be able to selectively schedule Profile tests to nodes that support
+   them (eg: Windows to windows nodes, GPU to nodes that have gpus, etc.)
+
```
Comment (spiffxp) [inline keps/sig-architecture/1618-conformance-profiles/README.md:273]: I'm assuming roles related to certification are going to be done as a followup

**Pull request merged**